### PR TITLE
Minor fixes for end device claiming with the CLI

### PR DESCRIPTION
**Pull request by KrishnaIyer**
#### Summary

Minor fixes for End Device claiming with the CLI Refs https://github.com/TheThingsNetwork/lorawan-stack/issues/6131
#### Changes

- Truly deprecate `with-claim-authentication-code`.
- Prevent claiming if there is no DevEUI.

#### Testing

Local

```
go run -tags=tti ./cmd/ttn-lw-cli dev create --with-claim-authentication-code
Flag --with-claim-authentication-code has been deprecated, Use a valid claim authentication code registered with a Join Server instead
```

##### Regressions

None. 

#### Notes for Reviewers

We don't actually currently deprecate a flag but forward it's contents to a replacement flag. This PR adds a function for "true" deprecation.

#### Checklist

- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [x] Compatibility: The changes are backwards compatible with existing API, storage, configuration and CLI, according to the compatibility commitments in `README.md` for the chosen target branch.
- [ ] Documentation: Relevant documentation is added or updated.
- [x] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [x] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -15,10 +15,14 @@ For details about compatibility between different releases, see the **Commitment
 
 ### Deprecated
 
+- `--with-claim-authentication-code` flag for the end device `create` command via the CLI. Users must use a valid claim authentication code that is registered on a Join Server instead of generating one during end device creation.
+
 ### Removed
 
 ### Fixed
 
+- Attempting to claim an end device with a generate DevEUI will now result in an error.
```
Comment (adriansmares) [inline CHANGELOG.md]: `generated`.

File: cmd/ttn-lw-cli/internal/util/flags.go [hunk 2]
```diff
@@ -40,6 +40,14 @@ func DeprecateFlag(flagSet *pflag.FlagSet, old string, new string) {
 	}
 }
 
+// DeprecateWithoutForwarding deprecates a CLI flag without forwarding it to a new flag.
+func DeprecateWithoutForwarding(flagSet *pflag.FlagSet, flag string, reason string) {
```
Comment (adriansmares) [inline cmd/ttn-lw-cli/internal/util/flags.go]: `DeprecateFlagWithoutForwarding`.

File: cmd/ttn-lw-cli/commands/end_devices.go [hunk 3]
```diff
@@ -518,13 +519,6 @@ var (
 					)
 				}
 			}
-			if withClaimAuthenticationCode, _ := cmd.Flags().GetBool("with-claim-authentication-code"); withClaimAuthenticationCode {
```
Comment (adriansmares) [inline cmd/ttn-lw-cli/commands/end_devices.go:521]: I think we should do this as part of https://github.com/TheThingsIndustries/lorawan-stack/issues/3036 , not necessarily now.

Comment (johanstokking) [reply cmd/ttn-lw-cli/commands/end_devices.go:521]: I agree. We should output a deprecation warning here.

File: cmd/ttn-lw-cli/commands/end_devices.go [hunk 4]
```diff
@@ -518,13 +519,6 @@ var (
...
-				device.ClaimAuthenticationCode = &ttnpb.EndDeviceAuthenticationCode{
-					Value: strings.ToUpper(hex.EncodeToString(random.Bytes(4))),
-				}
-				paths = append(paths, "claim_authentication_code")
-			}
-
 			claimOnExternalJS := len(device.ClaimAuthenticationCode.GetValue()) > 0
```
Comment (johanstokking) [inline cmd/ttn-lw-cli/commands/end_devices.go:537]: For now, this should include a condition that this CAC was not randomly generated just above. So claiming on an external JS is mutually exclusive with the deprecated CAC generation.

File: cmd/ttn-lw-cli/internal/util/flags.go [hunk 5]
```diff
@@ -40,6 +40,14 @@ func DeprecateFlag(flagSet *pflag.FlagSet, old string, new string) {
...
 }
 
+// DeprecateFlagWithoutForwarding deprecates a CLI flag without forwarding it to a new flag.
+func DeprecateFlagWithoutForwarding(flagSet *pflag.FlagSet, flag string, reason string) {
+	if flag := flagSet.Lookup(flag); flag != nil {
+		flag.Deprecated = reason
+		flag.Hidden = true
+	}
+}
+
```
Comment (KrishnaIyer) [inline cmd/ttn-lw-cli/internal/util/flags.go:50]: I'd like to still keep this function even if we are not using it now.

**Review by adriansmares — APPROVED** [review]

**Review by johanstokking — APPROVED** [review]

**Comment by KrishnaIyer** [comment]
Test flakes aren't relevant to the changes. Merging.

**Pull request merged**